### ISSUE TITLE
Tracee container fails to run on macos

### ISSUE THREAD

**Issue by georgettica** [aquasecurity/tracee#1127]
when I ran the commands:
```
$ docker run --name tracee --rm --pid=host --privileged -v /tmp/tracee:/tmp/tracee -v /lib/modules/:/lib/modules/:ro -v /usr/src:/usr/src:ro -it aquasec/tracee:latest
XXXX kernel headers could not be found, they are required for bpf compilation if CORE is not enabled. Set KERN_HEADERS to their path.
$ docker run --name tracee --rm --pid=host --privileged -v /tmp/tracee:/tmp/tracee -it aquasec/tracee:latest
XXXX kernel headers could not be found, they are required for bpf compilation if CORE is not enabled. Set KERN_HEADERS to their path.
Loaded 14 signature(s): [TRC-1 TRC-13  TRC-2 TRC-3 TRC-11 TRC-9 TRC-4 TRC-5 TRC-12 TRC-8 TRC-6 TRC-10 TRC-7]
```

I couldn't run tracee on my macos machine

``` 
$ docker version
Client:
 Cloud integration: 1.0.17
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.16.4
 Git commit:        f0df350
 Built:             Wed Jun  2 11:56:22 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       b0f5bc3
  Built:            Wed Jun  2 11:54:58 2021
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.4.6
  GitCommit:        d71fcd7d8303cbf684402823e425e9dd2e99285d
 runc:
  Version:          1.0.0-rc95
  GitCommit:        b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

```
$ sw_vers
ProductName:	macOS
ProductVersion:	11.6
BuildVersion:	20G165
$ uname -a
Darwin MacBook-Pro 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64 i386 MacBookPro16,1 Darwin
```

please let me know if I am missing something or there is more info I can provide

**Comment by georgettica** [comment]
as seen in https://github.com/aquasecurity/tracee/issues/737, there is a guide that seems to work on macos.. could the docs point to an official guide to run it?

**Comment by yanivagman** [comment]
I'm not a mac owner, but there was actually an official readme for this: https://github.com/aquasecurity/tracee/pull/571.
I see that it was removed by @itaysk .
@itaysk can you have a look?

**Comment by yanivagman** [comment]
@georgettica does following the instructions given in #570 helps?

**Comment by georgettica** [comment]
Nope, it seems that some commands have failed when trying to run a container with the kernel version (which is the latest)

**Comment by yanivagman** [comment]
Can you please provide the commands you tried to run with their output?

**Comment by georgettica** [comment]
Sure, that will take me a bit. But I'll do it :)

**Comment by georgettica** [comment]
so the issue was with the image being too low (5.10.25) I think
taking the latest and trying again

also I am trying to use `skopeo` instead of using tar multiple times to make the process cleaner (need to run skopeo in container)

**Comment by georgettica** [comment]
to record my steps.. the fastest way of getting the headers is:
```
docker run --name=data docker.io/linuxkit/kernel:5.12.14 ls
docker cp data:/kernel-dev.tar .
```

**Comment by yanivagman** [comment]
@georgettica so did you manage to run tracee on your mac?
@itaysk can we add this documentation back (#570 + #571) ?

**Comment by georgettica** [comment]
forgot about that for a bit.. the error I am getting now
```
XXX detected clang version: 11 is older than required minimum version: 12
```

but at least there is no issues with kernel headers :)

**Comment by georgettica** [comment]
odd
```
clang --version

Apple clang version 13.0.0 (clang-1300.0.29.3)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by yanivagman** [comment]
If your'e running tracee from a container (which you probably do), the clang used is of the tracee image and not the locally installed clang.
We recently had an issue with that which was fixed in main:
https://github.com/aquasecurity/tracee/pull/1173

We will release v0.6.5 soon that should solve this problem (meanwhile, you can try to build the tracee image locally using `make docker`)

**Comment by georgettica** [comment]
came back to the original error, trying to copy better now

**Comment by georgettica** [comment]
this seemed to work
```
docker run \
  --cgroupns=host \
  --name tracee \
  --pid=host \
  --privileged \
  --rm \
  -it \
  -v ${PWD}/usr/src/linux-headers-5.12.14-linuxkit/:/usr/src:ro \
  -v /lib/modules/:/lib/modules/:ro \
  -v /tmp/tracee:/tmp/tracee \
  tracee:latest
```
where I downloaded the package to a folder that is now `/usr/src` as it was auto-deleted

**Comment by simar7** [comment]
I just gave it a try and it seems to work for me (barring the clang issue, which will be fixed in v0.6.5 as @yanivagman mentioned).

I followed https://github.com/aquasecurity/tracee/pull/571/files as a guide. I think we should add them back.

**Comment by georgettica** [comment]
I'll try this aswell and hopefully this will work out of the box
